bok_choy/page_object.py

```diff
@@ -500,3 +520,130 @@ def wait_for_element_invisibility(self, element_selector, description, timeout=6
 
         """
         self.wait_for(lambda: not self.q(css=element_selector).visible, description=description, timeout=timeout)
+
+    def axs_audit_rules(self):
+        """
+        List of rules to check for accessibility errors on the page.
+        See https://github.com/GoogleChrome/accessibility-developer-tools/tree/master/src/audits
+
+        E.g. return ['badAriaAttributeValue']
+        An empty list means to check for all available rules.
+        None means that no audit should be done for this page.
+        """
+        return None
+
+    def axs_scope(self):
+        """
+        The "start point" for the audit: the element which contains the portion of
+        the page which should be audited.
+
+        E.g. return 'document.querySelector("div#foo")'
+        Defaults to using the document as the scope.
+        """
+        return 'null'
+
+    def do_axs_audit(self):
+        """
+        Use Google's Accessibility Developer Tools to audit the
+        page for accessibility problems.
+
+        See https://github.com/GoogleChrome/accessibility-developer-tools
+
+        Since this needs to inject JavaScript into the browser page, the only
+        known way to do this is to use PhantomJS as your browser.
+
+        Raises:
+            NotImplementedError if you are not using PhantomJS
+            RuntimeError if there was a problem getting the report
+
+        Returns:
+            A list (one for each browser session) of namedtuples with 'errors' and 'warnings'
+            fields whose values are the errors and warnings returned from the audit.
+
+            None if the page object has no rules defined to check.
+        """
+        if self.browser.name != 'phantomjs':
+            msg = 'Accessibility auditing is only supported with PhantomJS as the browser.'
+            raise NotImplementedError(msg)
+
+        rules = self.axs_audit_rules()
+        if rules is None:
```

Comment:
Sorry this is where I should have asked the question...should we raise an exception when no rules are defined, but someone's asking for an audit?